app/models/manageiq/providers/openstack/network_manager/refresh_parser.rb

```diff
@@ -204,12 +205,18 @@ def find_device_object(network_port, subnet_id)
       nil
     end
 
+    def parse_cloud_subnet_network_ports(cloud_subnet_network_port)
+      {
+        :address      => cloud_subnet_network_port["ip_address"],
+        :cloud_subnet => @data_index.fetch_path(:cloud_subnets, cloud_subnet_network_port["subnet_id"])
+      }
+    end
+
     def parse_network_port(network_port)
-      uid             = network_port.id
-      # There can be multiple fixed_ips on the port, but only under one subnet
-      subnet_id       = network_port.fixed_ips.try(:first).try(:[], "subnet_id")
-      device          = find_device_object(network_port, subnet_id)
-      security_groups = network_port.security_groups.blank? ? [] : network_port.security_groups.map do |x|
+      uid                        = network_port.id
+      cloud_subnet_network_ports = network_port.fixed_ips.map { |x| parse_cloud_subnet_network_ports(x) }
+      device                     = find_device_object(network_port)
+      security_groups            = network_port.security_groups.blank? ? [] : network_port.security_groups.map do |x|
```

Comment:
I don't think you need the `.blank?` check here at all, unless the security_groups can be `nil`?